crypto-app/src/pages/Coins/Coins.js

```diff
@@ -1,117 +1,238 @@
+import { TableContainer, TableHeader, Table, TableRow, SortButton } from './styles';
+
 
 class Coins extends React.Component {
   state = {
-    coins:[],
+    coins: [],
     page: 1,
     isLoading: false,
     hasMore: true,
     hasError: false,
-    errMess: ""
-  }
+    errMess: "",
+    sort: {
+      sortName: null,
+      sortPrice: null,
+      sortOneHour: null,
+      sortTwentyFourHour: null,
+      sortSevenDay: null
+    }
+  };
   getCoins = async () => {
     try {
       this.setState({
-        isLoading: true
-      })
+        isLoading: true,
+      });
       const { data } = await axios(
         `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
       );
       this.setState({
         coins: data,
         isLoading: false,
       });
-    } catch(error) {
+    } catch (error) {
       this.setState({
         isLoading: false,
         hasError: true,
-        errMess: "There was a problem getting coin list!"
-      })
+        errMess: "There was a problem getting coin list!",
+      });
     }
-  }
+  };
   getMoreCoins = async () => {
     try {
-      if (this.state.coins.length >= 1800) {
-        this.setState({ hasMore: false });
-        return;
-      }
+      const nextPage = this.state.page + 1
       this.setState({
-        page: this.state.page + 1,
+        page: nextPage,
         isLoading: true,
       });
-      const  { data }  = await axios(
-        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=${this.state.page}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+      console.log(this.state.page)
+      const { data } = await axios(
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=${nextPage}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
       );
-      const addCoins = [...this.state.coins, ...data]
-      setTimeout(() => {
-        this.setState({
-          coins: addCoins,
-          isLoading: false
-        })
-      }, 500)
-    }catch(error){
-      this.setState({isLoading: false, errMess:"Could not load more coins!"})
+      if(data === !![].length) {
+        this.setState({hasMore: false})
+      }
+      const addCoins = [...this.state.coins, ...data];
+      this.setState({
+        coins: addCoins,
+        isLoading: false,
+      });
+    } catch (error) {
+      this.setState({
+        isLoading: false,
+        errMess: "Could not load more coins!",
+      });
     }
-  }
+  };
   componentDidMount() {
-      this.getCoins()
+    this.getCoins();
   }
-  render(){
+  handleSort = (sortType) => {
+    const newSort = Object.entries(this.state.sort).map((entry) => {
+      const [key, value] = entry
+      if(key === sortType) {
+        return {
+          [key]: value !== true ? true : false 
```

Comment:
[key]: value !== true

Reply:
I'm not really clear on what you are asking me to do different here..